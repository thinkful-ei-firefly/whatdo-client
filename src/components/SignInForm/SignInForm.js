import React from 'react'
import AuthApiService from '../../services/auth-api-service'
import TokenService from '../../services/token-service'
import { UserContext } from '../../contexts/UserContext'
import { Link } from 'react-router-dom'
import LoadingLogin from '../LoadingLogin/LoadingLogin'
// import Loading from '../Loading/Loading'

export default class SignInForm extends React.Component {
  static contextType = UserContext
  state = { error: null, loading: false }

  handleSubmit = async e => {
    e.preventDefault()
    const { username, password } = e.target
    const loginCredentials = {
      username: username.value,
      password: password.value
    }

    try {
      this.setState({ loading: !this.state.loading })
      const token = await AuthApiService.login(loginCredentials)
      TokenService.saveAuthToken(token.authToken)
      // console.log('log in success!')
      // set context
      this.context.setUser({ username: username.value })
      // should update this to show a 'Success' message on next page
      this.setState({ loading: !this.state.loading })
      this.props.onSuccessfulSignIn()
    } catch (err) {
      this.setState({ error: err.error })
      this.setState({ loading: !this.state.loading })
    }
  }

  render() {
    const { error, loading } = this.state
    return <LoadingLogin />
    // if (loading) {
    //   return <LoadingLogin />
    // } else {
    //   return (
    //     <div className="SignInForm">
    //       <form className="box" onSubmit={this.handleSubmit}>
    //         <div className="alert ">{error && <p>{error}</p>}</div>
    //         <h1>Login</h1>
    //         <input
    //           className="txtb"
    //           type="text"
    //           autoComplete="off"
    //           id="username"
    //           name="username"
    //           required
    //           placeholder="Username"
    //         />
    //         {/**
    //   <label htmlFor="pass">Password</label>
    //   */}

    //         <input
    //           className="txtb"
    //           type="password"
    //           id="password"
    //           name="password"
    //           placeholder="Password"
    //           required
    //         />

    //         <input className="signin-btn" type="submit" value="Sign In" />
    //         <Link to="/signUp" className="haventAccYet">
    //           Donâ€™t have an account ?
    //         </Link>
    //       </form>
    //     </div>
    //   )
    // }
  }
}
