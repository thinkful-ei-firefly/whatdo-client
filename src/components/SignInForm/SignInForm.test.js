import React from 'react'
import ReactDOM from 'react-dom'
import SignInForm from './SignInForm'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

describe('<SignInForm/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <SignInForm />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <SignInForm />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
