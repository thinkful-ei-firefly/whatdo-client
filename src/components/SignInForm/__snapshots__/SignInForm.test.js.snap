// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm/> renders correctly 1`] = `
<div
  className="SignInForm"
>
  <form
    className="box"
    onSubmit={[Function]}
  >
    <div
      className="alert "
    />
    <h1>
      Login
    </h1>
    <input
      autoComplete="off"
      className="txtb"
      id="username"
      name="username"
      placeholder="Username"
      required={true}
      type="text"
    />
    <input
      className="txtb"
      id="password"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <input
      className="signin-btn"
      type="submit"
      value="Sign In"
    />
    <a
      className="haventAccYet"
      href="/signUp"
      onClick={[Function]}
    >
      Donâ€™t have an account ?
    </a>
  </form>
</div>
`;
